components/compiler/exprparser.cpp

```diff
@@ -324,6 +326,22 @@ namespace Compiler
                 mExplicit = name2;
                 return true;
             }
+
+            // In case someone tried passing an int in quotes
+            if (regex_match(name, std::regex("[+-]?[0-9]+")))
```

Comment:
Why don't just catch stof exception? If string has invalid format or overflow happens `stof` throws one. With current approach overflow is not handled. Same thing with `stoi`.

Reply:
IIRC, Morrowind in such case just uses the default value (0 for digits).
For example Morrowind treats the
`player->additem "steel longsword" "aaa"`
as a
`player->additem "steel longsword" 0`

Reply:
Which means that it really does use atof. Oh woe.

Reply:
How does the default value work? I just tried `player->setcurrentfatigue "shouldBeZero"` and vanilla thinks of it as a name and then does nothing rather than using 0.

![numberParsing](https://user-images.githubusercontent.com/23007670/63400632-5d042780-c389-11e9-9b37-68629e5231c1.jpg)

Reply:
In-game and TESCS compilers might be slightly different. Can you verify that TESCS isn't in fact able to compile a script with such an instruction?